web/js/animation/ui.js

```diff
@@ -350,9 +347,9 @@ wv.anim.ui = wv.anim.ui || function (models, ui) {
     var layer;
     var layers = models.layers.get();
 
-    for (var i = 0, len = layers; i < layers.length; i++) {
+    for (var i = 0, len = layers.length; i < len; i++) {
```

Comment:
Sure, but isn't the other implementation more efficient?

Reply:
No, `length` is a property of an array, it's not being calculated. `layers.length` is a pointer to a value, just like `len`, the only difference is if you do `len = layers.length` you're creating two variables in memory instead of just one. It's also simpler to read if there's no unneeded variable assignment. And unless we're running into performance problems, we should always prefer the most readable option;

> “Programs must be written for people to read, and only incidentally for machines to execute.”  ― Harold Abelson, [Structure and Interpretation of Computer Programs](https://mitpress.mit.edu/sicp/full-text/book/book.html)

> “Any fool can write code that a computer can understand. Good programmers write code that humans can understand.”  ― Martin Fowler, [Refactoring: Improving the Design of Existing Code](https://www.goodreads.com/book/show/44936.Refactoring)

Reply:
I disagree, we should default to the most performant method. In this case it would likely not make a difference, but with some of our `for loops` it could. It's better to be universally performant.

https://stackoverflow.com/a/7252102

Reply:
Interesting. That's something I didn't know. Thanks for sharing. I agree with you that in this case it likely won't make a difference, and I'd still prefer the more simple, readable option. But this is a trivial thing, so it doesn't really matter either way.